palette/src/component.rs

```diff
@@ -109,15 +119,61 @@ macro_rules! convert_float_to_uint {
                     #[inline]
                     fn into_component(self) -> $target {
                         let max = $target::max_intensity() as $temporary;
-                        let scaled = self as $temporary * max;
-                        clamp(scaled.round(), 0.0, max) as $target
+                        let scaled = (self as $temporary * max).min(max);
+                        let f = scaled + f64::from_bits(C52);
+                        (f.to_bits().saturating_sub(C52)) as  $target
                     }
                 }
             )+
         )*
     };
 }
 
+// Float to uint conversion with rounding to nearest even number. Formula
+// follows the form (x_f64 + C52_f64) - C52_u64, where x is the component.
+macro_rules! convert_double_to_uint {
+    ($float: ident; direct ($($direct_target: ident),+);) => {
```

Comment:
I guess `$float` is redundant here and above, now when they are split.

Reply:
I've changed that now :+1: , in the comment and macro